awxkit/awxkit/api/client.py

```diff
@@ -75,7 +75,7 @@ def request(self, relative_endpoint, method='get', json=None, data=None, query_p
 
         use_endpoint = relative_endpoint
         if self.server.endswith('/') and use_endpoint.startswith('/'):
-            raise RuntimeError('AWX URL given with trailing slash, remove slash.')
+            self.server = self.server[:-1]
```

Comment:
What if...

Some logic calls `connection.request` with relative_endpoint of `/api/v2/job_templates/` and then another method calls it with `api/v2/job_templates/`. Then you could wind up with `https://foo.com/api/v2jobtemplates/` maybe.

since we're getting into correcting errors of the data we're given, I would go further to strip the `/` and use a `'/'.join` to be sure.

This is going to matter more as we talk about having other code import awxkit and use its connection object to make requests. We can't count on any code consistency from what is calling these.

Reply:
made more generic